lib/connection.js

```diff
@@ -24,12 +24,53 @@ var EndpointState = require('./endpoint.js');
 var Session = require('./session.js');
 var Transport = require('./transport.js');
 
+var fs = require('fs');
+var os = require('os');
+var path = require('path');
 var net = require('net');
 var tls = require('tls');
 var EventEmitter = require('events').EventEmitter;
 
 var AMQP_PROTOCOL_ID = 0x00;
 
+function find_connect_config() {
+    var paths;
+    if (process.env.MESSAGING_CONNECT_FILE) {
+        paths = [process.env.MESSAGING_CONNECT_FILE];
+    } else {
+        paths = [process.cwd(), path.join(os.homedir(), '.config/messaging'),'/etc/messaging'].map(function (base) { return path.join(base, '/connect.json'); });
```

Comment:
It is worth documenting the order of precedence while considering the config file. What if, the config file `connect.json` is present in multiple locations? 
1. `<current working directory>/connect.json`
2. `<home dir>/.config/messaging/connect.json`
3.  `<home dir>/etc/messaging/connect.json`